app/models/notification.rb

```diff
@@ -15,7 +15,7 @@ def self.for(recipient, opts={})
   end
 
   def self.notify(recipient, target, actor)
-    return nil unless target.respond_to? :notification_type
+    return nil unless target.respond_to? :notification_type || recipient != actor
```

Comment:
@cmrd-senya is not able to reproduce the bug, @AugierLe42e you're the one who reported it so he's asking for a way to reproduce it. The perfect way to give the steps to reproduce would be a test. But you can also describe it here.